core/src/main/java/com/microsoft/applicationinsights/channel/concrete/inprocess/InProcessTelemetryChannel.java

```diff
@@ -47,313 +45,339 @@
-                                         LimitsEnforcer sendIntervalInSeconds,
-                                         boolean throttling) {
-        makeSureEndpointAddressIsValid(endpointAddress);
-
-        if (s_transmitterFactory == null) {
-            s_transmitterFactory = new InProcessTelemetryChannelFactory();
-        }
-
-        telemetriesTransmitter = s_transmitterFactory.create(endpointAddress, maxTransmissionStorageCapacity, throttling);
-        telemetryBuffer = new TelemetryBuffer(telemetriesTransmitter, maxTelemetryBufferCapacityEnforcer, sendIntervalInSeconds);
-
-        setDeveloperMode(developerMode);
-    }
-
-    /**
-     * The method will throw IllegalArgumentException if the endpointAddress is not a valid uri
-     * Please note that a null or empty string is valid as far as the class is concerned and thus considered valid
-     *
-     * @param endpointAddress
-     */
-    private void makeSureEndpointAddressIsValid(String endpointAddress) {
-        if (Strings.isNullOrEmpty(endpointAddress)) {
-            return;
-        }
-
-        URI uri = Sanitizer.sanitizeUri(endpointAddress);
-        if (uri == null) {
-            String errorMessage = String.format("Endpoint address %s is not a valid uri", endpointAddress);
-            InternalLogger.INSTANCE.error(errorMessage);
-            throw new IllegalArgumentException(errorMessage);
-        }
-    }
-
-    private LimitsEnforcer createDefaultMaxTelemetryBufferCapacityEnforcer(Integer currentValue) {
-        LimitsEnforcer maxItemsInBatchEnforcer =
-                LimitsEnforcer.createWithClosestLimitOnError(
-                        MAX_MAX_TELEMETRY_BUFFER_CAPACITY_NAME,
-                        MIN_MAX_TELEMETRY_BUFFER_CAPACITY,
-                        MAX_MAX_TELEMETRY_BUFFER_CAPACITY,
-                        DEFAULT_MAX_TELEMETRY_BUFFER_CAPACITY,
-                        currentValue == null ? DEFAULT_MAX_TELEMETRY_BUFFER_CAPACITY : currentValue);
-
-        return maxItemsInBatchEnforcer;
-    }
-
-    private LimitsEnforcer createDefaultSendIntervalInSecondsEnforcer(Integer currentValue) {
-        LimitsEnforcer sendIntervalInSecondsEnforcer =
-                LimitsEnforcer.createWithClosestLimitOnError(
-                        FLUSH_BUFFER_TIMEOUT_IN_SECONDS_NAME,
-                        MIN_FLUSH_BUFFER_TIMEOUT_IN_SECONDS,
-                        MAX_FLUSH_BUFFER_TIMEOUT_IN_SECONDS,
-                        DEFAULT_FLUSH_BUFFER_TIMEOUT_IN_SECONDS,
-                        currentValue == null ? DEFAULT_FLUSH_BUFFER_TIMEOUT_IN_SECONDS : currentValue);
-
-        return sendIntervalInSecondsEnforcer;
-    }
+	
+	private final static String INSTANT_RETRY_NAME = "MaxInstantRetry";
+	private final static int DEFAULT_MAX_INSTANT_RETRY = 3;
+	private final static int DEFAULT_MAX_TELEMETRY_BUFFER_CAPACITY = 500;
+	private final static int MIN_MAX_TELEMETRY_BUFFER_CAPACITY = 1;
+	private final static int MAX_MAX_TELEMETRY_BUFFER_CAPACITY = 1000;
```

Comment:
This is also definitely less. I think this should be definitely increased.

Reply:
It's the same as before, so we should not change it in this PR.